pkg/storegateway/partitioner.go

```diff
@@ -42,38 +43,41 @@ func newGapBasedPartitioner(maxGapBytes uint64, reg prometheus.Registerer) *gapB
 			Name: "cortex_bucket_store_partitioner_expanded_ranges_total",
 			Help: "Total number of byte ranges returned by the partitioner after they've been combined together to reduce the number of bucket API calls.",
 		}),
+		extendedRanges: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+			Name: "cortex_bucket_store_partitioner_extended_ranges_total",
+			Help: "Total number of byte ranges that were not overlapping but were joined because they were closer than the configured maximum gap.",
```

Comment:
[nit] Is my edit correct?

```suggestion
			Help: "Total number of byte ranges that were not adjacent or overlapping but were joined because they were closer than the configured maximum gap.",
```

Reply:
you are correct 👍

Reply:
given `[0, 10)` and `[10, 20)`  and gap size `0` the partitioner will merge these two

given `[0, 10)` and `[11, 20)`  and gap size `1` the partitioner will merge these two

given `[0, 10)` and `[12, 20)`  and gap size `1` the partitioner will not merge these two

i added a test case for the last two cases